### PR TITLE
Updated tests and code to crystal 0.25

### PR DESCRIPTION
**Pull request by Stygmates**
These are the changes needed to use crystal toml with the crystal 0.25 update, alongside the update of the tests for travis, thanks to @maiha for modifying the code [](https://github.com/maiha/crystal-toml/commit/9d0cac57853fda7f745cc42581ba34b4d9cc283f)

### CODE REVIEW

**Comment by Stygmates** [comment]
Thanks for your feedback, the changes have been merged. Is there anything else to look into/to pay attention to before merging this PR?

File: spec/cases_spec.cr [hunk 1]
```diff
@@ -37,6 +45,16 @@ private def compare(toml_hash : Hash, json_hash : Hash)
   end
 end
 
+private def compare(toml_array : Array, json : JSON::Any)
+  if json.raw.is_a?(Array)
+    compare toml_array, json.as_a
+  elsif json.raw.is_a?(Hash) && json["type"]? == "array" && json["value"]?
+    compare toml_array, json["value"].as_a
```
Comment (j8r) [inline spec/cases_spec.cr]: The json value can be reused
```crystal
  elsif json.raw.is_a?(Hash) && json["type"]? == "array" && (value = json["value"]?)
    compare toml_array, value.as_a
```

File: spec/lexer_spec.cr [hunk 2]
```diff
@@ -137,14 +137,14 @@ describe Lexer do
...
+  it_lexes_time "1979-05-27T07:32:00-07:30", Time.utc(1979, 5, 27, 15, 2, 0)
+  it_lexes_time "1979-05-27T07:32:00+07:30", Time.utc(1979, 5, 27, 0, 2, 0)
   it_lexes_time "1979-05-27T07:32:00.999999-07:00",
     {% if Crystal::VERSION =~ /^0\.(\d|1\d|2[0-3])\./ %}
-       Time.new(1979, 5, 27, 14, 32, 0, 999, kind: Time::Kind::Utc)
+       Time.utc(1979, 5, 27, 14, 32, 0, 999)
     {% else %}
-       Time.new(1979, 5, 27, 14, 32, 0, nanosecond: 999999000, kind: Time::Kind::Utc)
+       Time.utc(1979, 5, 27, 14, 32, 0, nanosecond: 999999000)
     {% end %}
```
Comment (j8r) [inline spec/lexer_spec.cr]: Not sure to still keep this ugly macros, especially we can't ensure the code will work on older Crystal compilers.

File: src/toml/lexer.cr [hunk 3]
```diff
@@ -477,8 +477,10 @@ class TOML::Lexer
     time =
       {% if Crystal::VERSION =~ /^0\.(\d|1\d|2[0-3])\./ %}
         Time.new(year, month, day, hour, minute, second, microseconds / 1000, kind: Time::Kind::Utc) # 0.23.x or lower
+      {% elsif Crystal::VERSION =~ /^0\.24\./ %}
+        Time.new(year, month, day, hour, minute, second, nanosecond: microseconds * 1000, kind: Time::Kind::Utc) # 0.24.x breaks arguments
       {% else %}
-        Time.new(year, month, day, hour, minute, second, nanosecond: microseconds * 1000, kind: Time::Kind::Utc) # 0.24.x or higher
+        Time.new(year.to_i32, month, day, hour, minute, second, nanosecond: microseconds * 1000, location: Time::Location::UTC) # 0.25.x breaks `Time::Kind`
       {% end %}
```
Comment (j8r) [inline src/toml/lexer.cr]: ditto

File: spec/cases_spec.cr [hunk 4]
```diff
@@ -1,31 +1,39 @@
...
 
-private def compare(toml : Time, json : Hash)
+private def compare(toml : Time, json : JSON::Any)
   json["type"].should eq("datetime")
   json["value"].should eq(toml.to_s("%Y-%m-%dT%H:%M:%SZ"))
 end
 
+private def compare(toml_hash : Hash, json : JSON::Any)
+  if json.as_h?
+    compare toml_hash, json.as_h
```
Comment (j8r) [inline spec/cases_spec.cr]: The hash can be reused.
```crystal
if hash = json.as_h?
  compare toml_hash, hash
```

**Comment by j8r** [comment]
The version has also to be bumped to 0.6.1.
The 0.6.0 tag is missing but referenced in the `shard.yml`.

**Comment by j8r** [comment]
Any update on this?

**Comment by j8r** [comment]
It would be good to also hear @asterite 's and @maiha 's opinion on this, being major contributors on this project. On my side I would be to only support the two latest major Crystal versions, rendering this macros obsolete.

**Comment by asterite** [comment]
Fork this project, I don't have time to maintain it. Once you fork it you can ask to delete this repo (my suggestion).

**Comment by j8r** [comment]
That's ok if @manastech doesn't have the time to maintain it. So I propose to fork the project to https://github.com/crystal-community . What do you think @maiha and @Stygmates ?
cc @veelenga , @hugoabonizio

**Comment by maiha** [comment]
Hi @j8r 
#### fork issue
agreed.

#### macros issue
I'd like to keep backward compatibility as much as possible. However, if it is the burden of maintainers, I agree to delete in the next version. 😸

**Comment by Stygmates** [comment]
The organization looks ok to me.

**Comment by veelenga** [comment]
I can grant access to move the repo. What should it be? @maiha?

**Comment by bcardiff** [comment]
I can transfer it to crystal-community so the shards that use this will keep working directly. Is that ok @veelenga / @maiha ?

**Comment by veelenga** [comment]
Sent invites to both of you. Should be good to go.

**Comment by bcardiff** [comment]
The repo has been transferred. I gave access to maintainers group. Thanks for the effort to keep toml up to date.

**Comment by veelenga** [comment]
@maiha sorry, added write access

**Comment by maiha** [comment]
Unfortunately I do not seem to have authority. Neither the merge button nor the setting tab is displayed.

**Comment by veelenga** [comment]
Added you to maintainers. Should be good now 😄

**Pull request merged**